lib/improver/lapse_rate.py

```diff
@@ -0,0 +1,402 @@
+        # before calculation.
+        y_data = temperature[~np.isnan(temperature)]
+        x_data = orography[~np.isnan(temperature)]
+
+        matrix = np.stack([x_data, np.ones(len(x_data))], axis=0).T
+        gradient, _ = lstsq(matrix, y_data)[0]
+
+        return gradient
+
+    def _create_heightdiff_mask(self, all_orog_subsections):
+        """
+        Function to create a mask for any neighbouring points where the height
+        difference from the central point is greater than max_height_diff.
+
+        Slice through the orography subsection array to remove central
+        points.
+        Extracts the height value of each central point and masks out the
+        neighbouring points where their height difference is greater than
+        the maximum.
+
+        Args:
+            all_orog_subsections(2D np.array):
+               Each row contains the height values of each neighbourhood.
+
+        Returns:
+            height_diff_mask (np.ndarray):
+                A 2D array of boolean values.
+
+        """
+
+        self.all_orog_subsections = all_orog_subsections
+
+        central_points = self.all_orog_subsections[:, self.ind_central_point]
+        central_points = np.swapaxes([central_points], 0, 1)
+
+        height_diff = np.subtract(self.all_orog_subsections, central_points)
+        height_diff = np.absolute(height_diff)
+
+        height_diff_mask = np.where(height_diff >= self.max_height_diff, True,
+                                    False)
+
+        return height_diff_mask
+
+    def process(self, temperature_cube, orography_cube, land_sea_mask_cube):
+        """Calculates the lapse rate from the temperature and orography cubes.
+
+        Args:
+            temperature_cube (iris.cube.Cube):
+                Cube of air temperatures (K).
+
+            orography_cube (Iris.cube.Cube):
+                Cube containing orography data (metres)
+
+            land_sea_mask_cube (iris.cube.Cube):
+                Cube containing a binary land-sea mask. True for land-points
+                and False for Sea.
+
+        Returns:
+            lapse_rate_cube (iris.cube.Cube):
+                Cube containing lapse rate (Km-1)
+
+        Raises
+        ------
+        TypeError: If input cubes are not cubes
+        ValueError: If input cubes are the wrong units.
+
+        """
+
+        if not isinstance(temperature_cube, iris.cube.Cube):
+            msg = "Temperature input is not a cube, but {}"
+            raise TypeError(msg.format(type(temperature_cube)))
+
+        if not isinstance(orography_cube, iris.cube.Cube):
+            msg = "Orography input is not a cube, but {}"
+            raise TypeError(msg.format(type(orography_cube)))
+
+        if not isinstance(land_sea_mask_cube, iris.cube.Cube):
+            msg = "Land/Sea mask input is not a cube, but {}"
+            raise TypeError(msg.format(type(land_sea_mask_cube)))
+
+        # Tests units are correct and raises error if not.
```

Comment:
Do we need to fail here?  Why can't we convert the input cubes into metres and kelvin?

Reply:
The reason I put in a fail was because I wanted to safeguard against the input cubes being in  entirely wrong units that couldn't be converted.

Reply:
You can convert the units on the input cube using eg `temperature_cube.convert_units('kelvin')`.  If the conversion cannot be made this will throw an error something like the following:
```
ValueError: Unable to convert from 'Unit('kelvin')' to 'Unit('metres')'
```

Reply:
Done!